File: Maximum-Product-of-Two-Digits.java
class Solution {
    public int maxProduct(int n) {
        ArrayList<Integer> a = new ArrayList<>();
        while (n > 0) {
            a.add(n % 10);
            n /= 10;
        }
        int s = a.size();
        int maxp = 0;
        for (int i = 0; i < s; i++) {
            for (int j = i + 1; j < s; j++) {
                maxp = Math.max(maxp, a.get(i) * a.get(j));
            }
        }
        return maxp;
    }
}

File: Maximum-Product-of-Two-Digits.cpp
class Solution {
public:
    int maxProduct(int n) {
        vector<int> v;
        while (n > 0) {
            v.push_back(n % 10);
            n = n/10;
        }
        int s = v.size();
        int maxp = 0;
        for (int i = 0; i < s; ++i) {
            for (int j = i + 1; j < s; ++j) {
                maxp = max(maxp, v[i] * v[j]);
            }
        }
        return maxp;
    }
};
